code/modules/admin/topic.dm

```diff
@@ -11,9 +11,29 @@
 		return TRUE
 	log_admin_private("[key_name(usr)] clicked an href with [msg] authorization key! [href]")
 
-/datum/admins/Topic(href, href_list)
-	..()
+// BASE ADMINS_TOPIC
+
+/datum/admins_topic()
+	var/keyword
+	var/log = TRUE
+	var/key_valid
+
+/datum/admins_topic/proc/TryRun(list/input)
+	key_valid = config && (CONFIG_GET(string/comms_key) == input["key"])

```

Comment:
You are re-using world_topic code here. The comms key is basically the password to use any of the world/Topic functionality. What you probably want to replace this with is `/datum/admins/proc/CheckAdminHref`.

Reply:
I noticed already and removed it locally. Just working on getting the first admins_topic datum made and testing it locally.
Thank you for stating that proc part. I will see about working that in later on. Right now I have it in admin/Topic still until this progresses further.